### PR TITLE
update rust toolchain version and crates version

### PR DESCRIPTION
**Pull request by LemonHX**
update rust toolchain to latest,
fix compiling errors.

Signed-off-by: lemonhx <lemonhx@lemonhx.tech>

### What is changed and how it works?

Close #xxx 

What's Changed:

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- PR to update `pingcap/tidb-ansible`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

```release-note
none
```

### CODE REVIEW

File: components/encryption/src/io.rs [hunk 1]
```diff
@@ -387,8 +387,9 @@ impl CrypterCore {
 
     fn reset_buffer(&mut self, size: usize) {
         // OCrypter require the output buffer to have block_size extra bytes, or it will panic.
-        self.buffer.reserve(size + self.block_size);
+        #[allow(clippy::uninit_vec)]
```
Comment (BusyJay) [inline components/encryption/src/io.rs:390]: I don't think it's a safe implementation. /cc @hunterlxt

File: components/external_storage/export/src/export.rs [hunk 2]
```diff
@@ -182,6 +182,7 @@ fn create_backend_inner(
         },
         #[cfg(not(any(feature = "cloud-gcp", feature = "cloud-aws")))]
         _ => return Err(bad_backend(backend.clone())),
+        Backend::AzureBlobStorage(_) => todo!(),
```
Comment (BusyJay) [inline components/external_storage/export/src/export.rs:185]: We use `unimplemented!` in TiKV.

**Review by BusyJay — COMMENTED** [review]
You should keep the crates version untouched. We don't upgrade dependencies unless necessary.

File: components/encryption/src/io.rs (hunk 1 above)
Comment (sticnarf) [reply components/encryption/src/io.rs:390]: Agree... Strictly speaking, every slice must be initialized. The problem is that the interface provided by the openssl library accepts a `& mut [u8]`. Passing an uninitialized buffer is UB but should work correctly in real world...

File: components/external_storage/export/src/export.rs (hunk 2 above)
Comment (LemonHX) [reply components/external_storage/export/src/export.rs:185]: ok get it

File: components/encryption/src/io.rs (hunk 1 above)
Comment (tabokie) [reply components/encryption/src/io.rs:390]: This is an output buffer, and uninit-ed vec is safe as long as it's not read before written to. Still, I'm not against using `resize` here, the perf difference should be negligible.

Comment (peng1999) [reply components/encryption/src/io.rs:390]: Using `Vec::set_len` like this is not panic-safe and should be avoided. The data should be written *before* the `Vec::set_len` call. See https://github.com/rust-lang/rust-clippy/issues/4483

Comment (tabokie) [reply components/encryption/src/io.rs:390]: I understand, but panic safety is only a blocker if we are storing value with non-trivial `Drop` implementation, which isn't the case here.

**Comment by sticnarf** [comment]
Hi, @LemonHX. Do you have time to continue this pull request?

**Comment by sticnarf** [comment]
Close due to inactivity.

**Pull request closed without merging**